buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1442,11 +1442,19 @@ private void checkReadableBytes0(int minimumReadableBytes) {
      * if the buffer was released before.
      */
     protected final void ensureAccessible() {
-        if (checkAccessible && refCnt() == 0) {
+        if (checkAccessible && internalRefCnt() == 0) {
```

Comment:
If this is best effort, why not just sample?  Maybe 1/16 times it checks.

Reply:
I think what we do now is better then sampling imho. so I would prefer to keep it as it is in the PR.